File: benchmarks/solutions/python/r1-word-freq.py
import sys
from collections import Counter

def main():
    try:
        text = open(sys.argv[1]).read().lower()
    except (IndexError, FileNotFoundError) as e:
        print(f"error: {e}", file=sys.stderr)
        sys.exit(1)

    words = [w for w in text.split() if w.isalpha()]
    counts = Counter(words)
    for word, count in counts.most_common(10):
        print(f"{word}: {count}")

if __name__ == "__main__":
    main()

File: benchmarks/solutions/swift/r1-word-freq.swift
import Foundation

guard CommandLine.arguments.count > 1 else {
    fputs("error: no file argument\n", stderr)
    exit(1)
}

guard let data = FileManager.default.contents(atPath: CommandLine.arguments[1]),
      let text = String(data: data, encoding: .utf8) else {
    fputs("error: cannot read file\n", stderr)
    exit(1)
}

var counts: [String: Int] = [:]
for word in text.lowercased().components(separatedBy: CharacterSet.letters.inverted) {
    if !word.isEmpty {
        counts[word, default: 0] += 1
    }
}

let sorted = counts.sorted { a, b in
    a.value != b.value ? a.value > b.value : a.key < b.key
}

for (word, count) in sorted.prefix(10) {
    print("\(word): \(count)")
}
